From 73c252b994d908c803aa50e60df8e3cf5e4c10b6 Mon Sep 17 00:00:00 2001
From: Ricardo Salveti <ricardo@foundries.io>
Date: Fri, 25 Jun 2021 12:26:44 -0300
Subject: [PATCH 090/233] [FIO extras] imx: mx7ulp: use mode register to verify
 if closed

Switch from using imx_hab_is_enabled (which requires IMX_HAB to be
enabled) to fetching SEC_CONFIG from the mode register instead.

Mode register provides BOOTCFG, and bit 17 can be used as SEC_CONFIG.

Signed-off-by: Ricardo Salveti <ricardo@foundries.io>
---
 arch/arm/mach-imx/mx7ulp/soc.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-imx/mx7ulp/soc.c b/arch/arm/mach-imx/mx7ulp/soc.c
index b53d9ada4c7..28af172559f 100644
--- a/arch/arm/mach-imx/mx7ulp/soc.c
+++ b/arch/arm/mach-imx/mx7ulp/soc.c
@@ -480,9 +480,13 @@ void boot_mode_enable_secondary(bool enable)
 	setbits_le32(&psrc->sim_dg0_ctrl1, WR_ACK_DGO_GP10);
 }
 
+#define CMC_MR_SEC_CONFIG (1 << 17)
+
 int boot_mode_is_closed(void)
 {
-	return imx_hab_is_enabled();
+	u32 *reg_mr = (u32 *)(SRC_BASE_ADDR + 0x40);
+
+	return !!(readl(reg_mr) & CMC_MR_SEC_CONFIG);
 }
 
 int boot_mode_getprisec(void)
-- 
2.40.1

